src/SessionGenerator.php

```diff
@@ -0,0 +1,303 @@
+     *
+     * @since [*next-version*]
+     *
+     * @return ValidatorInterface
+     */
+    protected function _getValidator()
+    {
+        return $this->validator;
+    }
+
+    /**
+     * Sets the validator.
+     *
+     * @since [*next-version*]
+     *
+     * @param ValidatorInterface $validator
+     *
+     * @return $this
+     */
+    protected function _setValidator(ValidatorInterface $validator = null)
+    {
+        $this->validator = $validator;
+
+        return $this;
+    }
+
+    /**
+     * Validates a session, if a validator is available.
+     *
+     * @since [*next-version*]
+     *
+     * @param mixed $session
+     *
+     * @return bool
+     */
+    protected function _validateSession($session)
+    {
+        $validator = $this->_getValidator();
+
+        if (is_null($validator)) {
+            return true;
+        }
+
+        try {
+            $validator->validate($session);
+
+            return true;
+        } catch (ValidationFailedExceptionInterface $e) {
+            return false;
+        }
+    }
+
+    /**
+     * Retrieves the session factory callback.
+     *
+     * @since [*next-version*]
+     *
+     * @return callable
+     */
+    protected function _getSessionFactory()
+    {
+        return $this->sessionFactory;
+    }
+
+    /**
+     * Sets the session factory callback.
+     *
+     * @since [*next-version*]
+     *
+     * @param callable $sessionFactory The session factory callback.
+     *
+     * @return $this
+     */
+    protected function _setSessionFactory(callable $sessionFactory)
+    {
+        $this->sessionFactory = $sessionFactory;
+
+        return $this;
+    }
+
+    /**
+     * Creates a session instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param int $start The start timestamp for the session.
+     * @param int $end   The end timestamp for the session.
+     *
+     * @return mixed The created session.
+     */
+    protected function _createSession($start, $end)
+    {
+        $factory = $this->_getSessionFactory();
+
+        return call_user_func_array($factory, [$start, $end]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function generate($start, $end)
+    {
+        $results = [];
+        $this->_generate($start, $end, $results);
+
+        return $results;
+    }
+
+    /**
+     * Generates sessions.
+     *
+     * @since [*next-version*]
+     *
+     * @param int   $rangeStart The range's starting timestamp.
+     * @param int   $rangeEnd   The range's ending timestamp.
+     * @param array $results    The results array that will be populated with the session instances.
+     * @param array $startTimes A temporary array for storing the starting times that have already been processed, to
+     *                          avoid repeating specific passes.
+     */
+    protected function _generate($rangeStart, $rangeEnd, array &$results = [], array &$startTimes = [])
```

Comment:
Looks like something that belongs in an abstract class.

Reply:
@mecha, also not addressed, it seems.

Reply:
You are commenting on code that doesn't exist anymore bro! 😆 This is outdated.